kedro-datasets/kedro_datasets/datasets/__init__.py

```diff
@@ -0,0 +1,3 @@
+"""``kedro_datasets.datasets`` is where you can find all of Kedro's data connectors.
```

Comment:
At a quick glance at [parent ticket](https://github.com/kedro-org/kedro/issues/1457), I don't see there are any ticket created for this. How are we going to link the documentation here to the `kedro` docs when we move dataset over here?

Reply:
I'm not sure I understand your question here? Do you mean we are missing a ticket to update all mentions in the docs from `kedro.extras.datasets` to `kedro_datasets.datasets`?

Reply:
@deepyaman We didn't think of this but it's a very interesting idea. Where are the places where that would be less breaking than what we're doing here? I can think of the case where a user defines a custom dataset that inherits from one of our built in ones (and hence does `from kedro.extras.datasets import ...`). Are there others?

What I think @noklam was asking about here though is about the documentation, i.e. how do we get [these dataset docs in the right place](https://kedro.readthedocs.io/en/stable/kedro.extras.datasets.html#module-kedro.extras.datasets). This is an excellent question and not something we'd thought of either. This actually sounds quite tricky and hard to do well 😬 @noklam please could you make a ticket for this if it's what you meant?

Reply:
@AntonyMilneQB It would also be less breaking in cases where people specify the full path to the dataset--which some users inexplicably do. Finally, for better or for worse, new datasets would potentially work out of the box, with partial paths, with older versions of Kedro (that don't have an updated default lookup path).

I think another part of it just comes down to whether you all think sharing the `kedro` namespace would be nice--the user feels like they're using one cohesive product--or confusing.

Reply:
@deepyaman All very good points I think. If we keep `kedro.extras.datasets` as a namespace then eventually I hope we could drop the superfluous `extras` part anyway rather than maintain it indefinitely for backwards compatibility.

Reply:
I think sharing the `kedro` namespace would be nice, although I do see why it would be potentially confusing also. The main motivation for splitting off `kedro-datasets` is for release/distribution purposes. It's still a part of kedro I think.

Reply:
@AntonyMilneQB @noklam I was just thinking about this more, in the context of something I'm working on--the alternative is to try and disassociate it from `kedro` as much as possible. Datasets don't actually use anything on the Kedro side, and I'm actually just now thinking about a case where I may just want the generic load/save abstraction that datasets provide. If I'm then leveraging this in a separate library, that really has nothing to do with `kedro`, it would feel awkward to be referencing the `kedro` namespace--I'd rather the top-level namespace just be called `datasets` (not even `kedro_datasets`).

Happy to talk this through, if it could be helpful for all involved; I think the tradeoffs are worth discussing. :)